components/trade-form/index.tsx

```diff
@@ -27,6 +28,36 @@ import { useExtrinsic } from "lib/hooks/useExtrinsic";
 import { useTradeItemState } from "lib/hooks/queries/useTradeItemState";
 import { calcInGivenOut, calcOutGivenIn, calcSpotPrice } from "lib/math";
 import TradeResult from "components/markets/TradeResult";
+import { TradeType } from "lib/types";
+
+const getTradeValuesFromExtrinsicResult = (
+  type: TradeType,
+  data: ISubmittableResult,
+): { baseAmount: string; assetAmount: string } => {
+  let baseAsset = new Decimal(0);
+  let outcome = new Decimal(0);
+  const { events } = data;
+  for (const eventData of events) {
+    const { event } = eventData;
+    const { data } = event;
+    if (
+      event.method === "SwapExactAmountIn" ||
+      event.method === "SwapExactAmountOut"
+    ) {
+      if (type === "buy") {
+        baseAsset = baseAsset.add(data[0]["assetAmountIn"].toPrimitive());
+        outcome = outcome.add(data[0]["assetAmountOut"].toPrimitive());
+      } else {
+        baseAsset = baseAsset.add(data[0]["assetAmountOut"].toPrimitive());
+        outcome = outcome.add(data[0]["assetAmountIn"].toPrimitive());
+      }
+    }
+  }
+  return {
+    baseAmount: baseAsset.div(ZTG).toFixed(4),
```

Comment:
ZTG is only worth $0.1 so 4 decimal places of precision is a bit excessive, probably 1 is plenty.

Reply:
they're in TradeResult - https://github.com/zeitgeistpm/ui/blob/b0352999a15222678e35232e78e754500e0ae688/components/trade-form/index.tsx#L183